### ISSUE TITLE
Accessibility: Cannot tab to or interact with files/headers in table view

### ISSUE THREAD

**Issue by ChloeSartorelli** [silverstripe/silverstripe-asset-admin#1611]
In the table view in asset admin (which displays files in a gridfield-like display), it's not possible to navigate to anything inside the actual table using keyboard. The tab order skips the table completely, going from the "change view" button directly to pagination.

This means we cannot do the following:
1. Select any file
2. Edit any file
3. Delete any file
4. Sort the table

I took a crack at making the checkbox and headers interactable, but Steve and I couldn't agree on what the correct behaviour should be. See https://github.com/silverstripe/silverstripe-asset-admin/pull/1614 for the discussion.

We need to discover what the correct keyboard interaction and screen reader behaviour should be for table view - which likely also means looking at what the behaviour should be in the gallery view so that the behaviour can be similar wherever that's sensible.

Note that the behaviour doesn't have to be identical between these two views - the fact that the information is being displayed differently and has different semantic HTML also makes it okay to change the interaction to suit the different display/semantics.

## Some options

1. [The listbox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/)? May not be suitable for tabular data
2. [The grid pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/)? Requires that keyboard focus can move to every single cell which may not be suitable
3. Make interaction the same as gridfields (i.e. add a `...` button to each row which when activated provides various actions that can be performed on that row
4. ??? Open to suggestions

## old PRs

- https://github.com/silverstripe/silverstripe-asset-admin/pull/1614
- https://github.com/silverstripe/developer-docs/pull/825

**Comment by emteknetnz** [comment]
Agreed offline with UX team that general approach is use tab-islands to quickly tab between components, then use arrows to navigate within the component. This would imply that only one of the files should be tabbable, and then the arrows keys are used to navigate between files.

The single file which is tabbable should probably dynamically change, it should be either:
- the first file in the table, or
- the file currently being edited, or
- the last file was selected using the arrows keys before tabbling out of the component
